lennoxs30api/s30api_async.py

```diff
@@ -481,7 +482,13 @@ async def messagePump(self) -> None:
                 "Accept-Encoding": "gzip, deflate"
                 #                'Accept-Encoding' : 'gzip, deflate'
             }
-            resp = await self.get(url, headers=headers)
+            params = {
+                "Direction": "Oldest-to-Newest",
+                "MessageCount": "10",
+                "StartTime": "1",
+                "LongPollingTimeout": "15"
```

Comment:
Good catch! I like pulling it out of this constant, so we can have greater flexibility at the point of the REST call.

Reply:
I just pulled this down locally to see if the cloud long poll would start working.  It does not:

This is with a 1 second scan.

2021-10-30 12:43:18 DEBUG (MainThread) [custom_components.lennoxs30] messagePump_task running
2021-10-30 12:43:18 DEBUG (MainThread) [lennoxs30api.s30api_async] {
    "messages": []
}
2021-10-30 12:43:19 DEBUG (MainThread) [custom_components.lennoxs30] messagePump_task running
2021-10-30 12:43:19 DEBUG (MainThread) [lennoxs30api.s30api_async] {
    "messages": []
}
2021-10-30 12:43:21 DEBUG (MainThread) [custom_components.lennoxs30] messagePump_task running
2021-10-30 12:43:21 DEBUG (MainThread) [lennoxs30api.s30api_async] {
    "messages": []
}
2021-10-30 12:43:22 DEBUG (MainThread) [custom_components.lennoxs30] messagePump_task running
2021-10-30 12:43:22 DEBUG (MainThread) [lennoxs30api.s30api_async] {
    "messages": []
}

Reply:
In the integration where we have this:

            if fast_polling == True:
                res = await asyncio.sleep(self._fast_poll_interval)
            else:
                res = await self.event_wait_mp_wakeup(self._poll_interval)

Are you running with zeros for the config?  I suppose we could default them to zero or skip this section of code if we are in local mode?

Reply:
Got away a bit fast, so since the Cloud long poll does not work - we will want cloud to run on a polling schedule whereas with local we should use long poll and reenter the long poll ASAP.

Reply:
Oh I didn't even realize there was a setting to adjust the polling intervals, I'll look into that!
This change should be very safe to merge for either situation. The local API may get the benefit of being on a long poll when a change comes through (perhaps with the reduced default poll length for local connections), and the cloud api will continue as behavior. The boolean will mesh well, allowing for triggering another (long) poll as soon as it returns with a result.
Glad to help!